hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-1.x-replication/src/test/java/com/google/cloud/bigtable/hbase1_x/replication/HbaseToCloudBigtableReplicationEndpointTest.java

```diff
@@ -414,4 +412,78 @@ public void testWriteFailureToBigtableDoesNotStallReplication()
         "Value mismatch", TestUtils.getValue(0), CellUtil.cloneValue(actualCells.get(1)));
     Assert.assertEquals(0, actualCells.get(1).getTimestamp());
   }
+
+  @Test
+  public void testMutationReplicationWithWALEntryFilter() throws IOException, InterruptedException {
+    Put put1 = new Put(FILTERED_ROW_KEY);
+    put1.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, 0, FILTERED_ROW_KEY);
+    put1.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER, 0, FILTERED_ROW_KEY);
+    hbaseTable.put(put1);
+
+    Put put2 = new Put(ROW_KEY);
+    put2.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    put2.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    hbaseTable.put(put2);
+
+    TestUtils.waitForReplication(
+        () -> {
+          //  replicate Entries will be zero
+          return TestReplicationEndpoint.replicatedEntries.get() >= 1;
+        });
+
+    Result cbtResult = cbtTable.get(new Get(FILTERED_ROW_KEY).setMaxVersions());
+    Result hbaseResult = hbaseTable.get(new Get(FILTERED_ROW_KEY).setMaxVersions());
+    Assert.assertTrue(cbtResult.isEmpty());
+    Assert.assertFalse(hbaseResult.isEmpty());
+    Assert.assertEquals(
+        "Number of cells , actual cells: " + hbaseResult.listCells(),
+        2,
+        2,
+        hbaseResult.listCells().size());
+
+    Result cbtResult2 = cbtTable.get(new Get(ROW_KEY).setMaxVersions());
+    Result hbaseResult2 = hbaseTable.get(new Get(ROW_KEY).setMaxVersions());
+    TestUtils.assertEquals(cbtResult2, hbaseResult2);
+  }
+
+  @Test
+  public void testReplicationWithScope() throws IOException, InterruptedException {
+    TableName tableName = TableName.valueOf(UUID.randomUUID().toString());
+    createTables(
+        tableName, HConstants.REPLICATION_SCOPE_GLOBAL, HConstants.REPLICATION_SCOPE_LOCAL);
+    hbaseTable = hbaseConnection.getTable(tableName);
+    cbtTable = cbtConnection.getTable(tableName);
+
+    Put put1 = new Put(ROW_KEY);
+    put1.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    put1.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    hbaseTable.put(put1);
+
+    TestUtils.waitForReplication(
+        () -> {
+          //  replicate Entries will be 1
+          return TestReplicationEndpoint.replicatedEntries.get() >= 1;
+        });
+
+    Result cbtResult = cbtTable.get(new Get(ROW_KEY).setMaxVersions());
+    Result hbaseResult = hbaseTable.get(new Get(ROW_KEY).setMaxVersions());
+    Assert.assertFalse(cbtResult.isEmpty());
+    Assert.assertFalse(hbaseResult.isEmpty());
+    Assert.assertEquals(
```

Comment:
this method seems weird, what are we validating ? The name should give me a hint, this is not an "assertEquals" method.

Reply:
Done.